### PR TITLE
Bm2 brid

### PR DESCRIPTION
**Pull request by AdithyaPetavue**
benchmark code for bird

### CODE REVIEW

File: benchmark_scripts/common_functions.py [hunk 1]
```diff
@@ -222,11 +253,33 @@ def initialize_files(model_file_path: str) -> Tuple[str, str]:
     return (output_file_path, metrics_file_path, log_file_path)
 
 
-def generate_gold_file(gold_file_list: Tuple[Any, Any], model_file_path: str) -> None:
+def generate_gold_file(gold_file_list: Tuple[Any, Any], model_file_path: str,inference_size: int) -> None:
     query_list, db_id_list = gold_file_list
     with open(f"{model_file_path}/gold.txt", "w") as f:
         for i in range(len(query_list)):
             f.write(f"{query_list[i]}\t{db_id_list[i]}\n\n")
+    pathForDevjson = f'../spider_data/bird_equal_split_{inference_size}.csv'
```
Comment (Siddarth-Petavue) [inline benchmark_scripts/common_functions.py:261]: change to `path_to_source_csv`

File: benchmark_scripts/common_functions.py [hunk 2]
```diff
@@ -283,7 +336,7 @@ def get_parsed_args(supported_models: Dict, host_env: str) -> Tuple[Any, Dict]:
         "--inst",
         dest="inst",
         type=str,
-        default="5,7,9,11,13",
+        default="0",
```
Comment (Siddarth-Petavue) [inline benchmark_scripts/common_functions.py:339]: get this from common_constant

File: benchmark_scripts/common_functions.py [hunk 3]
```diff
@@ -315,38 +368,17 @@ def get_parsed_args(supported_models: Dict, host_env: str) -> Tuple[Any, Dict]:
...
-        )
-        parser.add_argument(
-            "--best_of",
-            type=str,
-            default=4,
-            help="Name of the directory to store the compressed file",
-        )
-
     parsed_args = parser.parse_args()
-    parsed_args.target_dir = str(parsed_args.target_dir) + "/inference_results/"
```
Comment (Siddarth-Petavue) [inline benchmark_scripts/common_functions.py:343]: add this

File: benchmark_scripts/common_functions.py [hunk 4]
```diff
@@ -315,38 +368,17 @@ def get_parsed_args(supported_models: Dict, host_env: str) -> Tuple[Any, Dict]:
         "--target-directory",
         type=str,
         dest="target_dir",
-        default=CURRENT_FILE_PATH.parents[1],
+        default=CURRENT_FILE_PATH,
```
Comment (Siddarth-Petavue) [inline benchmark_scripts/common_functions.py:371]: revert this change

File: benchmark_scripts/common_functions.py [hunk 5]
```diff
@@ -315,38 +368,17 @@ def get_parsed_args(supported_models: Dict, host_env: str) -> Tuple[Any, Dict]:
...
-
     parsed_args = parser.parse_args()
-    parsed_args.target_dir = str(parsed_args.target_dir) + "/inference_results/"
 
     model_instructions = {}
     if parsed_args.model_instructions:
         for item in parsed_args.model_instructions.split("/"):
-            model_key, inst_list_string = item.split("=")
-            model_instructions[model_key] = [int(inst) for inst in inst_list_string.split(",")]
+            key, value = item.split("=")
```
Comment (Siddarth-Petavue) [inline benchmark_scripts/common_functions.py:380]: revert this change

File: benchmark_scripts/common_functions.py [hunk 6]
```diff
@@ -378,6 +410,12 @@ def intermediate_sql_match(sql_string: str) -> Tuple[bool, str]:
         return (True, match[0])
     else:
         return (False, processed_str)
+    
+
+
+
+
```
Comment (Siddarth-Petavue) [inline benchmark_scripts/common_functions.py:417]: remove unnecessary empty lines

File: benchmark_scripts/amazon-bedrock.py [hunk 7]
```diff
@@ -32,6 +32,10 @@
 supported_models = {
     "claude": BedrockModels.MODEL_ANTHROPIC_CLAUDE,
```
[the remark this answers is not in the document]
Comment (AdithyaPetavue) [reply benchmark_scripts/amazon-bedrock.py]: done

File: benchmark_scripts/amazon-bedrock.py [hunk 8]
```diff
@@ -32,6 +32,10 @@
 supported_models = {
     "claude": BedrockModels.MODEL_ANTHROPIC_CLAUDE,
     "llama": BedrockModels.MODEL_META_LLAMA,
+    "claude3-so":BedrockModels.MODEL_ANTHROPIC_CLAUDE_3_SONNET,
```
[the remark this answers is not in the document]
Comment (AdithyaPetavue) [reply benchmark_scripts/amazon-bedrock.py]: done

File: benchmark_scripts/amazon-bedrock.py [hunk 9]
```diff
@@ -71,23 +75,44 @@ def run_queries_on_bedrock(
             )
 
             prompt = system_prompt.replace("[context]", context).replace(
-                "[question]", question
-            )
-            if model_name == BedrockModels.MODEL_ANTHROPIC_CLAUDE:
+                    "[question]", "Question: "+question
+                ).replace("[hint]",str(evidence))
+            
+            if model_name == BedrockModels.MODEL_ANTHROPIC_CLAUDE_3_SONNET or model_name == BedrockModels.MODEL_ANTHROPIC_CLAUDE_3_HAIKU:
```
[the remark this answers is not in the document]
Comment (AdithyaPetavue) [reply benchmark_scripts/amazon-bedrock.py]: done

File: benchmark_scripts/amazon-bedrock.py [hunk 10]
```diff
@@ -71,23 +75,44 @@ def run_queries_on_bedrock(
...
-                "[question]", question
-            )
-            if model_name == BedrockModels.MODEL_ANTHROPIC_CLAUDE:
+                    "[question]", "Question: "+question
+                ).replace("[hint]",str(evidence))
+            
+            if model_name == BedrockModels.MODEL_ANTHROPIC_CLAUDE_3_SONNET or model_name == BedrockModels.MODEL_ANTHROPIC_CLAUDE_3_HAIKU:
+                
+                body = {
+                    "anthropic_version": "bedrock-2023-05-31",
```
[the remark this answers is not in the document]
Comment (AdithyaPetavue) [reply benchmark_scripts/amazon-bedrock.py]: done

File: benchmark_scripts/amazon-bedrock.py [hunk 11]
```diff
@@ -71,23 +75,44 @@ def run_queries_on_bedrock(
...
+                    "max_tokens": 300,
```
[the remark this answers is not in the document]
Comment (AdithyaPetavue) [reply benchmark_scripts/amazon-bedrock.py]: done

File: benchmark_scripts/amazon-bedrock.py [hunk 12]
```diff
@@ -71,23 +75,44 @@ def run_queries_on_bedrock(
...
+                body = {"prompt": prompt}
+                body["max_tokens_to_sample"] = 300
+            elif model_name == BedrockModels.MODEL_ANTHROPIC_MISTRAL_7B:
+                prompt = f"<s>[INST] {prompt}\n\n[/INST]"
+                body = {"prompt": prompt}
+                body["max_tokens"] = 300
+            elif model_name == BedrockModels.MODEL_ANTHROPIC_MIXTRAL:
+                prompt = f"<s>[INST]  {prompt}\n\n[/INST]"
+                body = {"prompt": prompt}
+                body["max_tokens"] = 300
```
[the remark this answers is not in the document]
Comment (AdithyaPetavue) [reply benchmark_scripts/amazon-bedrock.py]: done

File: benchmark_scripts/anthropic_runner.py [hunk 13]
```diff
@@ -0,0 +1,247 @@
...
+                }
+            ]
+        },
+    ]
+            data_to_log["request"] = req
+            response_time_start = datetime.now(timezone.utc)
+            anthropic_response = client.messages.create(
+                model=model_name,
+                messages=req,
+                max_tokens=1000,
```
[the remark this answers is not in the document]
Comment (AdithyaPetavue) [reply benchmark_scripts/anthropic_runner.py]: done

File: benchmark_scripts/common_constants.py [hunk 14]
```diff
@@ -37,6 +38,15 @@ class SelfHostedModels:
...
\ No newline at end of file
+    MODEL_WEIGHTS_DIRECTORY = "../model-weights/"
+
+class AnthropicModels:
+    MODEL_ANTHROPIC_OPUS = "claude-3-opus-20240229"
+    MODEL_ANTHROPIC_SONNET = "claude-3-sonnet-20240229"
+    MODEL_ANTHROPIC_HAIKU = "claude-3-haiku-20240307"
+
+class GeminiModels:
+    MODEL_GEMINI_PRO_1_5 = "models/gemini-1.0-pro-latest"
```
[the remark this answers is not in the document]
Comment (AdithyaPetavue) [reply benchmark_scripts/common_constants.py]: done

File: benchmark_scripts/common_functions.py (hunk 1 above)
Comment (AdithyaPetavue) [reply benchmark_scripts/common_functions.py]: done

File: benchmark_scripts/common_functions.py (hunk 2 above)
Comment (AdithyaPetavue) [reply benchmark_scripts/common_functions.py]: done

File: benchmark_scripts/common_functions.py (hunk 3 above)
Comment (AdithyaPetavue) [reply benchmark_scripts/common_functions.py:343]: done

File: benchmark_scripts/common_functions.py (hunk 4 above)
Comment (AdithyaPetavue) [reply benchmark_scripts/common_functions.py]: done

File: benchmark_scripts/common_functions.py (hunk 5 above)
Comment (AdithyaPetavue) [reply benchmark_scripts/common_functions.py:384]: done

File: benchmark_scripts/common_functions.py (hunk 6 above)
Comment (AdithyaPetavue) [reply benchmark_scripts/common_functions.py]: done

File: benchmark_scripts/amazon-bedrock.py [hunk 15]
```diff
@@ -108,7 +133,11 @@ def run_queries_on_bedrock(
 
             if model_name == BedrockModels.MODEL_ANTHROPIC_CLAUDE:
                 llm_response_content = response_body["completion"]
-            else:
+            elif model_name == BedrockModels.MODEL_ANTHROPIC_MISTRAL_7B or model_name == BedrockModels.MODEL_ANTHROPIC_MIXTRAL:
```
[the remark this answers is not in the document]
Comment (Siddarth-Petavue) [reply benchmark_scripts/amazon-bedrock.py]: Resolve

File: benchmark_scripts/common_constants.py [hunk 16]
```diff
@@ -10,25 +10,26 @@ class Environments:
...
     MODEL_META_LLAMA = "meta.llama2-70b-chat-v1"
     MODEL_ANTHROPIC_CLAUDE = "anthropic.claude-v2"
-
-
-class OpenAIModels:
-    MODEL_GPT_3 = "gpt-3.5-turbo-16k"
-    MODEL_GPT_4 = "gpt-4-turbo-preview"
+    MODEL_ANTHROPIC_CLAUDE_3_SONNET = "anthropic.claude-3-sonnet-20240229-v1:0"
+    MODEL_ANTHROPIC_CLAUDE_3_HAIKU = "anthropic.claude-3-haiku-20240307-v1:0"
+    MODEL_ANTHROPIC_MIXTRAL = "mistral.mixtral-8x7b-instruct-v0:1"
```
[the remark this answers is not in the document]
Comment (Siddarth-Petavue) [reply benchmark_scripts/common_constants.py:22]: Done
